### PR TITLE
Python: Add HF model and pipeline kwargs to allow model loading customization

### PR DESCRIPTION
**Pull request by karolzak**
### Motivation and Context

I was trying to use this library to load and use [HF Falcon model](https://huggingface.co/tiiuae/falcon-7b-instruct). Unfortunately in order to make it work I needed to pass in some additional arguments such as `torch_dtype`, `cache_dir` or `tokenizer`. It was not possible with current implementation so I decided to add `model_kwargs` and `pipeline_kwargs` to the `HuggingFaceTextCompletion` class init function so now, with the changes I introduced I can successfully load any `falcon` models by using this code:  
```python
import semantic_kernel as sk
import semantic_kernel.connectors.ai.hugging_face as sk_hf
import torch
from transformers import AutoTokenizer


cache_dir = "/mnt/tmp"
model_name = "tiiuae/falcon-7b-instruct"

tokenizer = AutoTokenizer.from_pretrained(
    pretrained_model_name_or_path=model_name,
    cache_dir=cache_dir,
    trust_remote_code=True)

hf_model = sk_hf.HuggingFaceTextCompletion(
    model_name,
    task="text-generation",
    model_kwargs={
        "cache_dir": cache_dir,
        "torch_dtype": torch.bfloat16
    },
    pipeline_kwargs={
        "tokenizer": tokenizer,
        "trust_remote_code": True,
        "device": 0,
    }
)
```

### Description

The changes I introduced add `model_kwargs` and `pipeline_kwargs` to `HuggingFaceTextCompletion` class so that these arguments are then passed to the underlying `transformers.pipeline()` function and thus allow more customization options.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by alexchaomander** [comment]
Thank you for this PR! Enabling open-source models is definitely a priority for us! Our team will review this soon

File: python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py [hunk 1]
```diff
@@ -26,6 +26,8 @@ def __init__(
         device: Optional[int] = -1,
         task: Optional[str] = None,
         log: Optional[Logger] = None,
+        model_kwargs: Dict[str, Any] = None,
```
Comment (lemillermicrosoft) [inline python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py:29]: @awharrison-28 @dmytrostruk  -- Would it be worth adding the falcon model to the integration tests?

Comment (karolzak) [reply python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py:29]: @awharrison-28 I sure can try, but falcon models are quite heavy though (even the Falcon-7B one) - you think the tests infra is heavy enough to carry it?

Comment (awharrison-28) [reply python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py:29]: The tests already use google/flan-t5-base, does this support cache_dir and torch_dtype?

Comment (karolzak) [reply python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py:29]: @awharrison-28 thanks!
I replaced part of the test code as you suggested but I'm struggling with linting on imports. My local black shows everything is fine but in the tests it fails.

Comment (awharrison-28) [reply python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py:29]: try `poetry run pre-commit run -c .conf/.pre-commit-config.yaml -a`

Comment (karolzak) [reply python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py:29]: My bad. It was `ruff`, not `black` linting issue. Fixing it now

**Review by lemillermicrosoft — APPROVED** [review]
Nice!

**Pull request merged**